### ISSUE TITLE
Listing DNS entries fails, expecting 2023-07-01-preview API version to be present.

### ISSUE THREAD

**Issue by JacekLakis-TomTom** [Azure/azure-cli#26813]
### Describe the bug

When trying to list A records inside my public DNS zone, Azure CLI 2.50.0 fails with invalid API versions. Works as expected for CLI version 2.49.0. Issue seems related with https://github.com/Azure/azure-cli/commit/6e905f4f63592098abf0e09b492f2f305cf60210 as I see `2023-07-01-preview` versions hardocded in some places.

### Related command

```bash
$ az network dns record-set a list -g public-dns -z mypubliczone.com
```

### Errors

```
(NoRegisteredProviderFound) No registered resource provider found for location 'global' and API version '2023-07-01-preview' for type 'dnszones/A'. The supported api-versions are '2015-05-04-preview, 2016-04-01, 2017-09-01, 2017-09-15-preview, 2017-10-01, 2018-03-01-preview, 2018-05-01'. The supported locations are ', global'.
Code: NoRegisteredProviderFound
Message: No registered resource provider found for location 'global' and API version '2023-07-01-preview' for type 'dnszones/A'. The supported api-versions are '2015-05-04-preview, 2016-04-01, 2017-09-01, 2017-09-15-preview, 2017-10-01, 2018-03-01-preview, 2018-05-01'. The supported locations are ', global'.
```

### Issue script & Debug output

I have following available versions of the `Microsoft.Network` provider for `dnszones/A` resource type:
```bash
$ az provider show --namespace Microsoft.Network --query "resourceTypes[?resourceType=='dnszones/A']"
[
  {
    "aliases": null,
    "apiProfiles": null,
    "apiVersions": [
      "2018-05-01",
      "2018-03-01-preview",
      "2017-10-01",
      "2017-09-15-preview",
      "2017-09-01",
      "2016-04-01",
      "2015-05-04-preview"
    ],
    "capabilities": "None",
    "defaultApiVersion": "2018-05-01",
    "locationMappings": null,
    "locations": [
      "global"
    ],
    "properties": null,
    "resourceType": "dnszones/A",
    "zoneMappings": null
  }
]
```
I tried re-registering provider in the subscription hoping that `2023-07-01-preview` will pop up, unfortunately no luck.

### Expected behavior

I can list my DNS zone A entries

### Environment Summary

azure-cli version 2.50.0

**Comment by yonzhan** [comment]
Thank you for opening this issue, we will look into it.

**Comment by JackTNeely** [comment]
Hello,

I am unable to export DNS zone files in the Azure PowerShell Cloud Shell and Azure Bash Cloud Shell in the Azure Portal.

The following commands fail in the Bash Cloud Shell:

1. `az network dns zone export -g myRg -n myZone -f myZone.txt`
2. `az network dns zone export -g myRg -n myZone -f /home/$USER/clouddrive/myZone.txt`

And this fails in the PowerShell Cloud Shell:

1. `az network dns zone export -g "myRg" -n "myZone" -f "\\home\\$USER\\clouddrive\\myZone.txt"`

This is the error produced (in both Shells):
`(NoRegisteredProviderFound) No registered resource provider found for location 'global' and API version '2023-07-01-preview' for type 'dnszones/recordsets'. The supported api-versions are '2015-05-04-preview, 2016-04-01, 2017-09-01, 2017-09-15-preview, 2017-10-01, 2018-03-01-preview, 2018-05-01'. The supported locations are ', global'.
Code: NoRegisteredProviderFound
Message: No registered resource provider found for location 'global' and API version '2023-07-01-preview' for type 'dnszones/recordsets'. The supported api-versions are '2015-05-04-preview, 2016-04-01, 2017-09-01, 2017-09-15-preview, 2017-10-01, 2018-03-01-preview, 2018-05-01'. The supported locations are ', global'.`

A quick run of `az provider show --namespace Microsoft.Network --query "registrationState"` **returns "Registered"**.

Please make note that this issue is occurring in the Cloud Shells.

Any workarounds to export a BIND-compatible zone file? This functionality is essential to operations. 
I might also add that Export should really be a button in the Azure Portal as well.

Thanks.

**Comment by alexftr** [comment]
That version 2.50.0 has been rolled out to Microsoft hosted agents on Azure DevOps and now many of our pipelines fails.
With message:
`ERROR: (NoRegisteredProviderFound) No registered resource provider found for location 'global' and API version '2023-07-01-preview' for type 'dnszones/CNAME'. The supported api-versions are '2015-05-04-preview, 2016-04-01, 2017-09-01, 2017-09-15-preview, 2017-10-01, 2018-03-01-preview, 2018-05-01'. The supported locations are ', global'.
Code: NoRegisteredProviderFound
Message: No registered resource provider found for location 'global' and API version '2023-07-01-preview' for type 'dnszones/CNAME'. The supported api-versions are '2015-05-04-preview, 2016-04-01, 2017-09-01, 2017-09-15-preview, 2017-10-01, 2018-03-01-preview, 2018-05-01'. The supported locations are ', global'.
`  

Downgrade is not an option because I need to do it in dozens of pipelines and it add 2.5 minutes extra time to run to each pipeline.

When do you plan to fix this issue.
Thanks.

**Comment by xorinzor** [comment]
Running into this same issue today on Azure Devops release pipeline using a `windows-2019` Azure Pipelines agent. Worked flawless daily before today.

**Comment by mischavandenburg** [comment]
Running into the same issue in a large enterprise organisation using Azure DevOps pipelines. We use Azure agents.

**Comment by marlowp** [comment]
We've encountered this issue kind of issue many times for various different CLI commands during the past 12 months (which is very disappointing). The following command works for downgrading the Azure CLI on ubuntu-20.04 Azure DevOps hosted agent:

```yaml
- task: PowerShell@2
  displayName: 'Revert to az cli version 2.49.0'
  inputs:
    targetType: 'inline'
    pwsh: true
    script: |
      pip install -Iv azure-cli==2.49.0 --extra-index-url https://azurecliprod.blob.core.windows.net/edge
```

**Comment by wx2486** [comment]
Got same issue while trying to create my personal site on Azure.

**Comment by KrysKruk** [comment]
Workaround for GitHub actions workflow:
```yaml
      - name: Install Azure cli
        run: |
          sudo apt-get install ca-certificates curl apt-transport-https lsb-release gnupg
          curl -sL https://packages.microsoft.com/keys/microsoft.asc | gpg --dearmor | sudo tee /etc/apt/trusted.gpg.d/microsoft.gpg > /dev/null
          AZ_REPO=$(lsb_release -cs)
          echo "deb [arch=`dpkg --print-architecture` signed-by=/etc/apt/trusted.gpg.d/microsoft.gpg] https://packages.microsoft.com/repos/azure-cli/ $AZ_REPO main" | sudo tee /etc/apt/sources.list.d/azure-cli.list
          sudo apt-get update
          sudo apt-get install --allow-downgrades azure-cli=2.49.0-1~jammy
```

Why won't you support two latest API versions or just keep the latest version in beta until all your products are ready?

**Comment by David-Noble-at-work** [comment]
> We've encountered this issue kind of issue many times for various different CLI commands during the past 12 months (which is very disappointing). The following command works for downgrading the Azure CLI on ubuntu-20.04 Azure DevOps hosted agent:
> 
> ```yaml
> - task: PowerShell@2
>   displayName: 'Revert to az cli version 2.49.0'
>   inputs:
>     targetType: 'inline'
>     pwsh: true
>     script: |
>       pip install -Iv azure-cli==2.49.0 --extra-index-url https://azurecliprod.blob.core.windows.net/edge
> ```

What's more disappointing is finding that a command you need (az postgres flexible-server) works in 2.50.0 and broken in 2.49.0. Now I must use an alternative workaround: a curl command, maybe. Anyone have any luck with that?

**Comment by wasker** [comment]
The biggest problem with the "downgrade" suggestion is this is impossible in Azure Cloud Shell.

**Comment by dominicusmento** [comment]
fix this asap

**Comment by justinmchase** [comment]
I was able to work around this by calling `az rest ...` and then the actual api endpoints. It was a pretty straight forwards replacement.

**Comment by florian-fischer-swarm** [comment]
Yes, this blocks several pipelines for us in devops (which no easy way to make the provided runners workable).

**Comment by maxburke** [comment]
How is this still a problem? Is there an ETA for a fix?

**Comment by buzztaiki** [comment]
This issue seems to be resolved in azure-cli 2.51.0.

```
❯❯ az version
{
  "azure-cli": "2.51.0",
  "azure-cli-core": "2.51.0",
  "azure-cli-telemetry": "1.1.0",
  "extensions": {
    "ml": "2.19.1",
    "reservation": "0.3.0"
  }
}
```

```
❯❯ az provider show --namespace Microsoft.Network --query "resourceTypes[?resourceType=='dnszones/A']"
[
  {
    "aliases": null,
    "apiProfiles": null,
    "apiVersions": [
      "2023-07-01-preview",
      "2018-05-01",
      "2018-03-01-preview",
      "2017-10-01",
      "2017-09-15-preview",
      "2017-09-01",
      "2016-04-01",
      "2015-05-04-preview"
    ],
    "capabilities": "None",
    "defaultApiVersion": "2018-05-01",
    "locationMappings": null,
    "locations": [
      "global"
    ],
    "properties": null,
    "resourceType": "dnszones/A",
    "zoneMappings": null
  }
]
```
